src/a01/runs.py

```diff
@@ -124,7 +126,7 @@ def create_run(image: str, from_failures: str = None, live: bool = False, parall
                                            'a01.reserved.fromrunfailure': from_failures,
                                        },
                                        details={
-                                           'a01.reserved.creator': get_user_id(),
+                                           'a01.reserved.creator': get_user_id() if email else username,
```

Comment:
What does the `get_user_id()` returns when the `a01` is logged in with service principal? Does that login have a name? If so we should use that. Otherwise, we should use the service principal name instead of a local `USERNAME`

The logic behind this is that no matter user specify `--email` or not, it should try to find the user identifier through the JWT token first since it is the name provided by the authentication authority and carries more weight then the local `USERNAME` environment variable.

Reply:
The only fields in the service principal token are `servicePrincipalId`, `servicePrincipalTenant` and `accessToken`.
There are a lot more fields in the user token, including `userId`.

Reply:
Still I can look into using the service principal name :)

Reply:
Yes. Service Principal name is helpful. It can help us to identify what's the account trigger the run.

Reply:
Fixed! :D